### ISSUE TITLE
AllInstanceConfig should be created in PersistentPreRun

### ISSUE THREAD

**Issue by LalatenduMohanty** [minishift/minishift#1460]
As part of profiles feature we are creating the AllInstanceConfig file in init() of root.go [1]. which is wrong in some way as ideally it should be done in PersistentPreRun to keep it consistent. Hence we are getting https://issues.jboss.org/browse/CDK-180
[1] https://github.com/minishift/minishift/blob/master/cmd/minishift/cmd/root.go#L217

**Comment by gbraad** [comment]
Not sure why this would triggers a "delete"...
```
$ ./minishift setup-cdk
Setting up CDK 3 on host using '/home/amit/.minishift' as Minishift's home directory
The MINISHIFT_HOME directory '/home/amit/.minishift' exists. Continuing will delete any existing VM and all other data in this directory. Do you want to continue? [y/N]
```

**Comment by LalatenduMohanty** [comment]
@gbraad It is because the directory is created before setup-cdk is run and if setup-cdk sees .minishift it will ask confirmation to overwrite it.

**Comment by LalatenduMohanty** [comment]
Resolved by #1465
